ngraph_bridge/ngraph_capture_variables.cc

```diff
@@ -40,6 +41,83 @@ static bool IsOutputNode(const Node* node,
   return found;
 }
 
+Status ReplacePrefetch(Graph* graph, Node* prefetch_node) {
+  NodeBuilder::NodeOut input_dataset;
+  NodeBuilder::NodeOut buffer_size;
+
+  std::vector<const Edge*> input_edges;
+  TF_RETURN_IF_ERROR(prefetch_node->input_edges(&input_edges));
+
+  input_dataset =
+      NodeBuilder::NodeOut(input_edges[0]->src(), input_edges[0]->src_output());
+  buffer_size =
+      NodeBuilder::NodeOut(input_edges[1]->src(), input_edges[1]->src_output());
+
+  std::vector<DataType> output_types;
+  TF_RETURN_IF_ERROR(
+      GetNodeAttr(prefetch_node->attrs(), "output_types", &output_types));
+
+  std::vector<PartialTensorShape> output_shapes;
+  TF_RETURN_IF_ERROR(
+      GetNodeAttr(prefetch_node->attrs(), "output_shapes", &output_shapes));
+
+  int slack_period = 0;
+  TF_RETURN_IF_ERROR(
+      GetNodeAttr(prefetch_node->attrs(), "slack_period", &slack_period));
+
+  Node* replacement;
+  TF_RETURN_IF_ERROR(NodeBuilder("NGraphPrefetchNode", "NGraphPrefetchDataset")
+                         .Input(input_dataset)
+                         .Input(buffer_size)
+                         .Attr("output_types", output_types)
+                         .Attr("output_shapes", output_shapes)
+                         .Attr("slack_period", slack_period)
+                         .Device(prefetch_node->assigned_device_name())
+                         .Finalize(graph, &replacement));
+  replacement->set_assigned_device_name(prefetch_node->assigned_device_name());
+
+  string new_name = graph->NewName("NGraph" + prefetch_node->name());
+  replacement->set_name(new_name);
+
+  std::vector<const Edge*> edges;
+
+  // Remove all the input edges of the existing prefetch node
+  std::vector<const Edge*> edges_to_remove;
+  std::vector<std::tuple<Node*, int, Node*, int>> edges_to_add;
+  for (auto edge : prefetch_node->in_edges()) {
+    edges_to_remove.push_back(edge);
+  }
+
+  for (auto edge : prefetch_node->out_edges()) {
+    NGRAPH_VLOG(0) << "Replacing: OutEdge " << edge->DebugString();
+    // Collect new output edges between the new prefetch node and the next node
+    edges_to_add.push_back(std::tuple<Node*, int, Node*, int>(
+        replacement, edge->src_output(), edge->dst(), edge->dst_input()));
+    // Remove the output edge from the current prefetch node
+    edges_to_remove.push_back(edge);
+  }
+
+  // Now add the new output edges
+  // The input edges to the new node is added during the node creation
+  for (const auto& i : edges_to_add) {
+    NGRAPH_VLOG(0) << "Adding: " << get<0>(i)->name() << "  " << get<1>(i)
```

Comment:
This comment is not for Avijit but for rest of us:
store values of theses get<0>.....get<I>, into variables,
and then use them.